### ISSUE TITLE
OpenApi specification using path variable with dash is not recognised

### ISSUE THREAD

**Issue by matejkobza** [outofcoffee/imposter#432]
Hello, I have discovered that in case we are using dash in path variable inside of OAS file it is not recognized

**OAS:**
```yaml
openapi: "3.0.1"
info:
  version: "v1"
  title: test
  description: 'testing'
paths:
  /test/{id-test}:
    get:
      parameters:
        - name: id-test
          in: path
          required: true
          schema:
            type: integer
            format: int64
      responses:
        "200":
          description: OK
          content:
            application/json:
              schema:
                type: string
              example: |-
                {
                  "hello": "world"
                }
```

**Config:**
`
{
  "plugin": "openapi",
  "specFile": "test.yaml"
}
`

**Response:**
<img width="807" alt="Screenshot 2023-06-29 at 14 26 15" src="https://github.com/outofcoffee/imposter/assets/3455842/d6fa5aa4-8274-4227-8e9b-edd378090bf3">

**When modified path variable without dash:**
```yaml
openapi: "3.0.1"
info:
  version: 'v1'
  title: test
  description: 'testing'

paths:
  /test/{id}:
    get:
      parameters:
        - name: id
          in: path
          required: true
          schema:
            type: integer
            format: int64
      responses:
        '200':
          description: OK
          content:
            application/json:
              schema:
                type: string
              example: |-
                {
                  "hello": "world"
                }
```

**Response is successful:**
<img width="879" alt="Screenshot 2023-06-29 at 14 38 55" src="https://github.com/outofcoffee/imposter/assets/3455842/7baa8610-407c-46f6-ae65-7338fc8abf89">

**Comment by outofcoffee** [comment]
Hi @matejkobza - thank you for raising this.

Looks like we might need to update the regex that detects OpenAPI format path parameters. I can’t seem to find a canonical source for what characters are valid. Any suggestions other than adding dash?

**Comment by hilsonp** [comment]
Maybe this is if some help... or not 😉

https://github.com/OAI/OpenAPI-Specification/issues/1969

**Comment by outofcoffee** [comment]
Thanks very much for the references @hilsonp.

Digging into this, the issue is with the conversion of OpenAPI path params (which are permissive) to those in Vert.x Web. [The documentation](https://vertx.io/docs/vertx-web/java/#_capturing_path_parameters) says:

> The placeholders consist of `:` followed by the parameter name. Parameter names consist of any alphabetic character, numeric character or underscore.

This could be a bit tricky, as if we normalise them to get them into Vert.x, we'll have to denormalise them so they can be referred to from matchers, scripts etc.

**Comment by austen-herbst** [comment]
@outofcoffee Hello! I'm evaluating Imposter for our use case, but came across this issue. We internally have a standard where all path parameter names must be "param case" or "kebab case" (lower case separated by dashes). So we have many OpenAPI specs that will face this issue.

Do you know if a fix for this bug is on the roadmap in the near future?

**Comment by outofcoffee** [comment]
This should be fixed in [v3.38.3](https://github.com/outofcoffee/imposter/releases).

Please let us know if this resolves your issue, or re-open if not.

**Comment by austen-herbst** [comment]
Yes, that fixed it! Thank you @outofcoffee!
